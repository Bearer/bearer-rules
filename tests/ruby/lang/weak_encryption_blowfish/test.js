const { createInvoker, getEnvironment } = require("../../../helper.js")
const { ruleId, ruleFile, testBase } = getEnvironment(__dirname)

describe(ruleId, () => {
  const invoke = createInvoker(ruleId, ruleFile, testBase)
  

  test("blowfish", () => {
    const testCase = "blowfish.rb"
    expect(invoke(testCase)).toMatchSnapshot();
  })
  

  test("blowfish_data", () => {
    const testCase = "blowfish_data.rb"
    expect(invoke(testCase)).toMatchSnapshot();
  })
  
})