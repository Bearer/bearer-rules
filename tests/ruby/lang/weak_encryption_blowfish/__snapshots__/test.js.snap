// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ruby_lang_weak_encryption_blowfish blowfish 1`] = `
"{
  "low": [
    {
      "cwe_ids": [
        "327"
      ],
      "id": "ruby_lang_weak_encryption_blowfish",
      "title": "Usage of weak encryption algorithm (Blowfish)",
      "description": "## Description\\n\\nA weak encryption library can lead to data breaches and greater security risk.\\n\\n## Remediations\\nAccording to [OWASP](https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/09-Testing_for_Weak_Cryptography/04-Testing_for_Weak_Encryption), Blowfish is considered a weak encryption algorithm and therefore shouldn't be used.\\n\\n❌ Avoid libraries and algorithms with known weaknesses:\\n\\n\`\`\`ruby\\nCrypt::Blowfish.new(\\"weak password encryption\\")\\n\`\`\`\\n\\n✅ Instead, we recommend using bcrypt:\\n\\n\`\`\`ruby\\nBCrypt::Password.create('iLOVEdogs123')\\n\`\`\`\\n\\n## Resources\\n- [BCrypt Explained](https://dev.to/sylviapap/bcrypt-explained-4k5c)\\n",
      "documentation_url": "https://docs.bearer.com/reference/rules/ruby_lang_weak_encryption_blowfish",
      "line_number": 2,
      "full_filename": "/tmp/bearer-scan/blowfish.rb",
      "filename": ".",
      "source": {
        "start": 2,
        "end": 4,
        "column": {
          "start": 1,
          "end": 4
        }
      },
      "sink": {
        "start": 2,
        "end": 4,
        "column": {
          "start": 1,
          "end": 4
        },
        "content": "blowfish.encrypt_block do\\n  \\"hello world\\"\\nend"
      },
      "parent_line_number": 2,
      "snippet": "blowfish.encrypt_block do\\n  \\"hello world\\"\\nend",
      "fingerprint": "bd49baf006f18f3010498485ae3f5dfa_0",
      "old_fingerprint": "5648e1c4bb9d2084f25ff405cf6fb9cd_0",
      "code_extract": "blowfish.encrypt_block do\\n  \\"hello world\\"\\nend"
    }
  ]
}"
`;

exports[`ruby_lang_weak_encryption_blowfish blowfish_data 1`] = `
"{
  "high": [
    {
      "cwe_ids": [
        "327"
      ],
      "id": "ruby_lang_weak_encryption_blowfish",
      "title": "Usage of weak encryption algorithm (Blowfish)",
      "description": "## Description\\n\\nA weak encryption library can lead to data breaches and greater security risk.\\n\\n## Remediations\\nAccording to [OWASP](https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/09-Testing_for_Weak_Cryptography/04-Testing_for_Weak_Encryption), Blowfish is considered a weak encryption algorithm and therefore shouldn't be used.\\n\\n❌ Avoid libraries and algorithms with known weaknesses:\\n\\n\`\`\`ruby\\nCrypt::Blowfish.new(\\"weak password encryption\\")\\n\`\`\`\\n\\n✅ Instead, we recommend using bcrypt:\\n\\n\`\`\`ruby\\nBCrypt::Password.create('iLOVEdogs123')\\n\`\`\`\\n\\n## Resources\\n- [BCrypt Explained](https://dev.to/sylviapap/bcrypt-explained-4k5c)\\n",
      "documentation_url": "https://docs.bearer.com/reference/rules/ruby_lang_weak_encryption_blowfish",
      "line_number": 1,
      "full_filename": "/tmp/bearer-scan/blowfish_data.rb",
      "filename": ".",
      "data_type": {
        "category_uuid": "14124881-6b92-4fc5-8005-ea7c1c09592e",
        "name": "Fullname"
      },
      "category_groups": [
        "PII",
        "Personal Data"
      ],
      "source": {
        "start": 2,
        "end": 2,
        "column": {
          "start": 3,
          "end": 17
        }
      },
      "sink": {
        "start": 1,
        "end": 3,
        "column": {
          "start": 1,
          "end": 2
        },
        "content": "Crypt::Blowfish.new(\\"insecure\\").encrypt_block { |user|\\n  user.full_name\\n}"
      },
      "parent_line_number": 1,
      "snippet": "Crypt::Blowfish.new(\\"insecure\\").encrypt_block { |user|\\n  user.full_name\\n}",
      "fingerprint": "bd49baf006f18f3010498485ae3f5dfa_0",
      "old_fingerprint": "425bf82b0c8dade97640afdaae8f6567_0",
      "code_extract": "Crypt::Blowfish.new(\\"insecure\\").encrypt_block { |user|\\n  user.full_name\\n}"
    },
    {
      "cwe_ids": [
        "327"
      ],
      "id": "ruby_lang_weak_encryption_blowfish",
      "title": "Usage of weak encryption algorithm (Blowfish)",
      "description": "## Description\\n\\nA weak encryption library can lead to data breaches and greater security risk.\\n\\n## Remediations\\nAccording to [OWASP](https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/09-Testing_for_Weak_Cryptography/04-Testing_for_Weak_Encryption), Blowfish is considered a weak encryption algorithm and therefore shouldn't be used.\\n\\n❌ Avoid libraries and algorithms with known weaknesses:\\n\\n\`\`\`ruby\\nCrypt::Blowfish.new(\\"weak password encryption\\")\\n\`\`\`\\n\\n✅ Instead, we recommend using bcrypt:\\n\\n\`\`\`ruby\\nBCrypt::Password.create('iLOVEdogs123')\\n\`\`\`\\n\\n## Resources\\n- [BCrypt Explained](https://dev.to/sylviapap/bcrypt-explained-4k5c)\\n",
      "documentation_url": "https://docs.bearer.com/reference/rules/ruby_lang_weak_encryption_blowfish",
      "line_number": 5,
      "full_filename": "/tmp/bearer-scan/blowfish_data.rb",
      "filename": ".",
      "data_type": {
        "category_uuid": "14124881-6b92-4fc5-8005-ea7c1c09592e",
        "name": "Fullname"
      },
      "category_groups": [
        "PII",
        "Personal Data"
      ],
      "source": {
        "start": 6,
        "end": 6,
        "column": {
          "start": 3,
          "end": 17
        }
      },
      "sink": {
        "start": 5,
        "end": 7,
        "column": {
          "start": 1,
          "end": 4
        },
        "content": "Crypt::Blowfish.new(\\"insecure\\").encrypt_block do |user|\\n  user.full_name\\nend"
      },
      "parent_line_number": 5,
      "snippet": "Crypt::Blowfish.new(\\"insecure\\").encrypt_block do |user|\\n  user.full_name\\nend",
      "fingerprint": "bd49baf006f18f3010498485ae3f5dfa_1",
      "old_fingerprint": "425bf82b0c8dade97640afdaae8f6567_1",
      "code_extract": "Crypt::Blowfish.new(\\"insecure\\").encrypt_block do |user|\\n  user.full_name\\nend"
    },
    {
      "cwe_ids": [
        "327"
      ],
      "id": "ruby_lang_weak_encryption_blowfish",
      "title": "Usage of weak encryption algorithm (Blowfish)",
      "description": "## Description\\n\\nA weak encryption library can lead to data breaches and greater security risk.\\n\\n## Remediations\\nAccording to [OWASP](https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/09-Testing_for_Weak_Cryptography/04-Testing_for_Weak_Encryption), Blowfish is considered a weak encryption algorithm and therefore shouldn't be used.\\n\\n❌ Avoid libraries and algorithms with known weaknesses:\\n\\n\`\`\`ruby\\nCrypt::Blowfish.new(\\"weak password encryption\\")\\n\`\`\`\\n\\n✅ Instead, we recommend using bcrypt:\\n\\n\`\`\`ruby\\nBCrypt::Password.create('iLOVEdogs123')\\n\`\`\`\\n\\n## Resources\\n- [BCrypt Explained](https://dev.to/sylviapap/bcrypt-explained-4k5c)\\n",
      "documentation_url": "https://docs.bearer.com/reference/rules/ruby_lang_weak_encryption_blowfish",
      "line_number": 9,
      "full_filename": "/tmp/bearer-scan/blowfish_data.rb",
      "filename": ".",
      "data_type": {
        "category_uuid": "cef587dd-76db-430b-9e18-7b031e1a193b",
        "name": "Email Address"
      },
      "category_groups": [
        "PII",
        "Personal Data"
      ],
      "source": {
        "start": 9,
        "end": 9,
        "column": {
          "start": 48,
          "end": 58
        }
      },
      "sink": {
        "start": 9,
        "end": 9,
        "column": {
          "start": 1,
          "end": 59
        },
        "content": "Crypt::Blowfish.new(\\"your-key\\").encrypt_string(user.email)"
      },
      "parent_line_number": 9,
      "snippet": "Crypt::Blowfish.new(\\"your-key\\").encrypt_string(user.email)",
      "fingerprint": "bd49baf006f18f3010498485ae3f5dfa_2",
      "old_fingerprint": "425bf82b0c8dade97640afdaae8f6567_2",
      "code_extract": "Crypt::Blowfish.new(\\"your-key\\").encrypt_string(user.email)"
    }
  ]
}"
`;
